src/components/mods/RandomHeldItemsMod.ts

```diff
@@ -0,0 +1,197 @@
+                "probability": .005
+            }
+        ],
+        "Flying": [
+            {
+                "name": "TM43",
+                "probability": .005
+            }
+        ],
+        "Grass": [
+            {
+                "name": "Leaf Stone",
+                "probability": .005
+            }
+        ],
+        "Poison": [
+            {
+                "name": "Antidote",
+                "probability": .05
+            },
+            {
+                "name": "TM06",
+                "probability": .005
+            }
+        ],
+        "Electric": [
+            {
+                "name": "Thundertone",
+                "probability": .005
+            }
+        ],
+        "Ground": [
+            {
+                "name": "Guard Spec",
+                "probability": .005
+            }
+        ],
+        "Psychic": [
+            {
+                "name": "Ultra Ball",
+                "probability": .005
+            }
+        ],
+        "Rock": [
+            {
+                "name": "Escape Rope",
+                "probability": .05
+            },
+            {
+                "name": "Iron",
+                "probability": .005
+            }
+        ],
+        "Ice": [
+            {
+                "name": "Ice Heal",
+                "probability": .025
+            },
+            {
+                "name": "TM13",
+                "probability": .005
+            }
+        ],
+        "Bug": [
+            {
+                "name": "Repel",
+                "probability": .025
+            },
+            {
+                "name": "Max Repel",
+                "probability": .005
+            }
+        ],
+        "Dragon": [
+            {
+                "name": "Rare Candy",
+                "probability": .0025
+            }
+        ],
+        "Ghost": [
+            {
+                "name": "Full Restore",
+                "probability": .0025
+            }
+        ]
+     };
+
+     /**
+      * Name of the mod.
+      */
+     public readonly name: string = "Random Held Items";
+
+     /**
+      * Mod events, keyed by name.
+      */
+     public readonly events: ICallbackRegister = {
+         [this.eventNames.onWildGrassPokemonChosen]: (chosenInfo: INewPokemon) => {
+             const pokemonName: string = chosenInfo.title.join("");
+             const pokemonType: string = this.gameStarter.constants.pokemon.byName[pokemonName].types[0];
+             const chosenItem = this.randomHeldItemGenerator(chosenInfo, pokemonType);
+
+             if (chosenItem[0] !== "") {
+                 chosenInfo.item = chosenItem;
+             }
+         }
+     };
+
+     /**
+      * Chooses which item is chosen for onNewPokemonCreation.
+      *
+      * @param chosenInfo   Info chosen by chooseRandomWildPokemon.
+      * @param pokemonType   Type of the wild encountered Pokemon.
+      * @returns Returns the name of an item or "" if no item generated.
```

Comment:
+1 to returning ASAP. But, it would be cleaner to return `undefined` if nothing is generated. That's both semantically closer (we're returning nothing, not some agreed-upon constant) and easier to work with in code (we can use the compiler to know it's either a name or nothing).